common/models/middleware-phase.js

```diff
@@ -31,7 +33,13 @@ module.exports = function(MiddlewarePhase) {
       }
       const id = filter.where.id;
       const workspace = WorkspaceManager.getWorkspace(options.workspaceId);
-      middlewareHandler.findPhase(workspace, id, cb);
+      workspace.Phase.refresh(workspace, id, function(err, results) {
+        if (err) return cb(err);
+        const phase = workspace.getMiddlewarePhase(id);
+        const middleware = phase.getMiddlewareList();
+        if (middleware) return cb(null, middleware);
+        cb('middleware not found');
```

Comment:
In fact, refactor this to a shared function because the cb is idential to MiddlewarePhase.all